Data_Package_Documentation/IGSN_Formatting/recreate_SESAR_template.R

```diff
@@ -0,0 +1,393 @@
+  
+})
+
+
+test_that("Required SITES columns are present", {
+  
+  required_cols <- c("Sample Name", "Material", "Collector/Chief Scientist", 
+                     "Collection date", "Collection method description", "Field program/cruise",
+                     "Latitude", "Longitude", "Primary physiographic feature", "Release Date")
+  
+  expect_true(all(required_cols %in% colnames(output_samples_template)))
+  
+  cat("Required cols NOT in your template: ")
+  cat("\n")
+  print(setdiff(required_cols, colnames(output_samples_template)))
+  cat("\n")
+  cat("Non-required cols in your template: ")
+  cat("\n")
+  print(setdiff(colnames(output_samples_template), required_cols))
+  cat("\n")
+  cat("\n")
+
+})
+
+
+test_that("Required SAMPLES columns are present", {
+  
+  required_cols <- c("Sample Name", "Material", "Collector/Chief Scientist", 
+                     "Collection date", "Collection method description", "Field program/cruise",
+                     "Latitude", "Longitude", "Primary physiographic feature", "Release Date")
+  
+  expect_true(all(required_cols %in% colnames(output_sites_template)))
+  
+  cat("Required cols NOT in your template: ")
+  cat("\n")
+  print(setdiff(required_cols, colnames(output_sites_template)))
+  cat("\n")
+  cat("Non-required cols in your template: ")
+  cat("\n")
+  print(setdiff(colnames(output_sites_template), required_cols))
+  cat("\n")
+  cat("\n")
+  
+})
+
+
+### Create readme for ESS_DIVE_Infrastructure_ONLY folder ######################
+
+readme <- c(
+  paste0(format(Sys.Date(), "%d %B %Y")),
+  
+  "River Corridor Science Focus Area",
+  "Pacific Northwest National Lab",
+  "",
+  "The files in this folder are for the purpose of ESS-DIVE infrastructure ONLY and can be ignored by data users. These files and associated column headers are not described anywhere in the data package. For a description of these files, go to https://ess-dive.gitbook.io/sample-id-and-metadata/.",
+  "",
+  'If you have looked at the data in this folder and are interested in the contents, you can find the same information within the main data package folder in a machine-readable file ending in "IGSN-Mapping.csv."'
+)
+
+readme
+
+
+### Write out files ############################################################
+
+# write out readme
+write_lines(readme, paste0(ess_dive_infrastructure_only_dir, "/readme_ESS-DIVE_Insfrastructure_ONLY.md"))
+
+# write out sites
+write_csv(output_site_header, paste0(ess_dive_infrastructure_only_dir, "/igsn_metadata_sites_for_samples.csv"))
+
+write_csv(output_sites_template, paste0(ess_dive_infrastructure_only_dir, "/igsn_metadata_sites_for_samples.csv"), append = TRUE, col_names = TRUE, na = "")
+
+# write out samples
+write_csv(output_sample_header, paste0(ess_dive_infrastructure_only_dir, "/igsn_metadata_samples.csv"))
+
+write_csv(output_samples_template, paste0(ess_dive_infrastructure_only_dir, "/igsn_metadata_samples.csv"), append = TRUE, col_names = TRUE, na = "")
```

Comment:
Does anything need to be done after? what if a column is empty?

Reply:
also waiting on ess-dive clarification for how to handle the final template output